romancal/dark_current/dark_current_step.py

```diff
@@ -90,9 +92,9 @@ def save_dark_data_as_dark_model(dark_data, dark_model):
 
     # Create DarkRef object and copy dark data to it
     out_dark = testutil.mk_dark(shape=dark_data.data.shape)
-    out_dark.data = dark_data.data
+    out_dark.data = u.Quantity(dark_data.data, out_dark.data.unit, dtype=out_dark.data.dtype)
     out_dark.dq = dark_data.groupdq
-    out_dark.err = dark_data.err
+    out_dark.err = u.Quantity(dark_data.err, out_dark.err.unit, dtype=out_dark.err.dtype)
```

Comment:
Can these changes be instead pushed to DarkCurrentStep.process(...) after dark_sub.do_correction?

Reply:
I think it may be cleaner where it presently is (the out & dark are split up this way) - potentially moreso with changes from #644.